engine-tests/src/test/java/org/terasology/config/flexible/SettingTest.java

```diff
@@ -41,7 +39,7 @@
 
         @Before
         public void setUp() {
-            setting = new SettingImpl<>(SETTING_ID,
+            setting = new SettingImpl<>(Integer.class, SETTING_ID,
```

Comment:
@eviltak, you mentioned removing the need for the class parameter by checking on the type of the default value? Can you please do that?